src/main/scala/carldata/series/TimeSeries.scala

```diff
@@ -470,5 +470,30 @@ case class TimeSeries[V](idx: Vector[LocalDateTime], ds: Vector[V]) {
     new TimeSeries(joinR(dataPoints, ts.dataPoints))
   }
 
+  /** Outer join. If one of the series doesn't have a data point then put default(left or right) value. */
+  def joinOuter[U](ts: TimeSeries[U], defaultLeft: V, defaultRight: U): TimeSeries[(V, U)] = {
+    val builder: mutable.ListBuffer[(LocalDateTime, (V, U))] = ListBuffer()
+
+    @tailrec def joinR(dp1: Vector[(LocalDateTime, V)],
```

Comment:
I think it is better to name params xs and ys instead of dp1 and dp2
And then x instead of p1.